src/Dapr.Jobs/DaprJobsClient.cs

```diff
@@ -0,0 +1,123 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using Dapr.Jobs.Models.Responses;
+
+namespace Dapr.Jobs;
+
+/// <summary>
+/// <para>
+/// Defines client operations for managing Dapr jobs.
+/// Use <see cref="DaprJobsClientBuilder"/> to create a <see cref="DaprJobsClient"/> or register
+/// for use with dependency injection via <see>
+///     <cref>DaprJobsServiceCollectionExtensions.AddDaprJobsClient</cref>
+/// </see>.
+/// </para>
+/// <para>
+/// Implementations of <see cref="DaprJobsClient"/> implement <see cref="IDisposable"/> because the
+/// client accesses network resources. For best performance, create a single long-lived client instance
+/// and share it for the lifetime of the application. This is done for you if created via the DI extensions. Avoid
+/// creating a disposing a client instance for each operation that the application performs - this can lead to socket
+/// exhaustion and other problems.
+/// </para>
+/// </summary>
+public abstract class DaprJobsClient : IDisposable
+{
+    private bool disposed;
+    
+    /// <summary>
+    /// Schedules a recurring job using a cron expression.
+    /// </summary>
+    /// <param name="jobName">The name of the job being scheduled.</param>
+    /// <param name="cronExpression">The systemd Cron-like expression indicating when the job should be triggered.</param>
+    /// <param name="startingFrom">The optional point-in-time from which the job schedule should start.</param>
+    /// <param name="repeats">The optional number of times the job should be triggered.</param>
+    /// <param name="ttl">Represents when the job should expire. If both this and DueTime are set, TTL needs to represent a later point in time.</param>
+    /// <param name="payload">The main payload of the job.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+    public abstract Task ScheduleCronJobAsync(string jobName, string cronExpression, DateTime? startingFrom = null,
```

Comment:
I don't recall if we discussed it before, but I believe we should prefer (as per the [docs](https://learn.microsoft.com/en-us/dotnet/standard/datetime/choosing-between-datetime#the-datetimeoffset-structure)) `DateTimeOffset` to `DateTime`.

Reply:
I need to get in a better habit of doing this myself. Fixed.